llvm/lib/Transforms/IPO/ArgumentPromotion.cpp

```diff
@@ -437,6 +440,13 @@ static bool allCallersPassValidPointerForArgument(Argument *Arg,
   // direct callees.
   return all_of(Callee->users(), [&](User *U) {
     CallBase &CB = cast<CallBase>(*U);
+    if (RecursiveCalls.contains(&CB))
+      return true;
+
+    // if (RecursiveCalls.size() &&
+    //     CB.getCalledFunction()->getName() == Callee->getName())
+    //   return true;
```

Comment:
Commented out code

Reply:
Yes, I will remove it. Was referring to it, in my thread with efriedma!